source/ota.c

```diff
@@ -2567,6 +2572,39 @@ static IngestResult_t processDataBlock( OtaFileContext_t * pFileContext,
     return eIngestResult;
 }
 
+/* Store the data block in file context. */
+static IngestResult_t writeDataBlockToFile( OtaFileContext_t * pFileContext,
```

Comment:
Is a separate function really necessary here?

Reply:
If I add an assertion in original processDataBlock function, the complexity level is 9.
So I add this separate function to reduce the level.

Reply:
If I add an assertion in original processDataBlock function, the complexity level is 9.
So I add this separate function to reduce the level.

Reply:
I pushed some changes to address this. Please look them over.

Reply:
Thanks. That's great!
I also added an unit test case for the coverage.